libbpf-rs/src/map.rs

```diff
@@ -734,3 +738,121 @@ impl<'a> Iterator for MapKeyIter<'a> {
         }
     }
 }
+/// Represent a wrapper `bpf_map_info` struct pointer
+/// Provide ability to retrive the details of a certain map with its fd only
+#[derive(Debug)]
+pub struct MapInfo {
+    /// The inner bpf_map_info object
+    pub info: bpf_map_info,
+}
+
+impl MapInfo {
+    /// Create a MapInfo struct from a fd.
+    pub fn new(fd: i32) -> Result<Self> {
+        // SAFETY: create a zeroed C stryct, and call a FFI function to fill it.
+        let mut map_info = unsafe { mem::zeroed::<bpf_map_info>() };
+        let mut size = mem::size_of_val(&map_info) as u32;
+        // SAFETY: This function fills `map_info` with at most `size` bytes.
+        let ret = unsafe {
+            bpf_obj_get_info_by_fd(
+                fd,
+                &mut map_info as *mut bpf_map_info as *mut c_void,
+                &mut size as *mut u32,
+            )
+        };
+        if ret < 0 {
+            return Err(Error::System(errno::errno()));
+        }
+        if size != mem::size_of_val(&map_info) as _ {
+            return Err(Error::Internal(format!(
+                "Unexpected info size: {} expected {}",
+                size,
+                mem::size_of_val(&map_info)
+            )));
+        }
+        Ok(Self { info: map_info })
+    }
+    /// Get the map type
+    pub fn map_type(&self) -> MapType {
+        match MapType::try_from(self.info.type_) {
+            Ok(t) => t,
+            Err(_) => MapType::Unknown,
+        }
+    }
+    /// Get the map id
+    pub fn id(&self) -> u32 {
+        self.info.id
+    }
```

Comment:
Do we still need all those getters now that `info` is public?

Reply:
I removed the single-line getter now